backend/app/bug_handler_test.go

```diff
@@ -0,0 +1,402 @@
+package app
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"path"
+	"testing"
+
+	"github.com/codescalersinternships/Flyspray/models"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateBug(t *testing.T) {
+	tempDir := t.TempDir()
+	path := path.Join(tempDir, "flyspray.db")
+	gin.SetMode(gin.TestMode)
+
+	// Create a new instance of your App
+	app, err := NewApp(path)
+	assert.NoError(t, err)
+
+	// Create a new HTTP request
+	router := gin.Default()
+	// middleware to set the "user_id" in the gin context
+	router.Use(func(c *gin.Context) {
+		c.Set("user_id", "99")
+		c.Next()
+	})
+	router.POST("/bug", WrapFunc(app.createBug))
+
+	t.Run("create new bug successfully", func(t *testing.T) {
+		bugInput := createBugInput{
+			Summary:     "hello world!",
+			ComponentID: 13,
+		}
+
+		payload, err := json.Marshal(bugInput)
+		if err != nil {
+			t.Fatal("failed to marshal bug payload")
+		}
+
+		req, err := http.NewRequest("POST", "/bug", bytes.NewBuffer(payload))
+		assert.NoError(t, err)
+
+		recorder := httptest.NewRecorder()
+
+		router.ServeHTTP(recorder, req)
+
+		assert.Equal(t, http.StatusCreated, recorder.Code, "got %d status code but want status code 201", recorder.Code)
```

Comment:
check also that bug is added to db, and for all tests check that changes are reflected in db